CDCgov/prime-simplereport
# Disable download button for over 20k test results

# FRONTEND PULL REQUEST

## Related Issue
Resolves #4061

## Changes Proposed
- on the frontend, prevent users from downloading more than 20,000 test results
- creates a separate component for the CSV download modal
- adds test for CSV download modal

## Additional Information
adding Better Data teammates as reviewers since G&E is running at lower capacity 😅 

## Testing
- Have an org with over 20k tests or change the `totalEntries` value being passed to the `DownloadResultsCsvModal` component to some number above 20,000 in the `frontend/src/app/testResults/DownloadResultsCsvButton.tsx` file
- Click "Download results"
- You should see the new copy below:
<img width="675" alt="Screen Shot 2022-07-26 at 5 30 11 PM" src="https://user-images.githubusercontent.com/20211771/181295736-c90ea496-a951-43a4-87ac-b1c2eadcdd90.png">

- Try downloading results for 20k or under test results with or without filters
- You should be able to download a csv as intended
 
## Screenshots / Demos

https://user-images.githubusercontent.com/20211771/181299940-7c2228dc-2a1f-4ea5-bb57-ae7af52f2683.mov



## Checklist for Author and Reviewer

### Design

### Content

### Support

### Security

### Documentation
